thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/datalayer/bao/DAOTestBase.java

```diff
@@ -0,0 +1,279 @@
+    // create schema
+    URL createSchemaUrl = getClass().getResource("/schema/create-schema.sql");
+    ScriptRunner scriptRunner = new ScriptRunner(conn, false, false);
+    scriptRunner.setDelimiter(";", true);
+    scriptRunner.runScript(new FileReader(createSchemaUrl.getFile()));
+  }
+
+  private void cleanUpJDBC() throws Exception {
+    System.out.println("Cleaning database: start");
+    try (Connection conn = ds.getConnection()) {
+      URL deleteSchemaUrl = getClass().getResource("/schema/drop-tables.sql");
+      ScriptRunner scriptRunner = new ScriptRunner(conn, false, false);
+      scriptRunner.runScript(new FileReader(deleteSchemaUrl.getFile()));
+    }
+    new File(dbUrlId).delete();
+    System.out.println("Cleaning database: done!");
+  }
+
+  @Override
+  public AnomalyFunctionManager getAnomalyFunctionDAO() {
+    return anomalyFunctionDAO;
+  }
+
+  @Override
+  public RawAnomalyResultManager getRawAnomalyResultDAO() {
+    return rawAnomalyResultDAO;
+  }
+
+  @Override
+  public JobManager getJobDAO() {
+    return jobDAO;
+  }
+
+  @Override
+  public TaskManager getTaskDAO() {
+    return taskDAO;
+  }
+
+  @Override
+  public MergedAnomalyResultManager getMergedAnomalyResultDAO() {
+    return mergedAnomalyResultDAO;
+  }
+
+  @Override
+  public DatasetConfigManager getDatasetConfigDAO() {
+    return datasetConfigDAO;
+  }
+
+  @Override
+  public MetricConfigManager getMetricConfigDAO() {
+    return metricConfigDAO;
+  }
+
+  @Override
+  public DashboardConfigManager getDashboardConfigDAO() {
+    return dashboardConfigDAO;
+  }
+
+  @Override
+  public OverrideConfigManager getOverrideConfigDAO() {
+    return overrideConfigDAO;
+  }
+
+  @Override
+  public AlertConfigManager getAlertConfigDAO() {
+    return alertConfigDAO;
+  }
+
+  @Override
+  public DataCompletenessConfigManager getDataCompletenessConfigDAO() {
+    return dataCompletenessConfigDAO;
+  }
+
+  @Override
+  public EventManager getEventDAO() {
+    return eventDAO;
+  }
+
+  @Override
+  public DetectionStatusManager getDetectionStatusDAO() {
+    return detectionStatusDAO;
+  }
+
+  @Override
+  public AutotuneConfigManager getAutotuneConfigDAO() {
+    return autotuneConfigDAO;
+  }
+
+  @Override
+  public ClassificationConfigManager getClassificationConfigDAO() {
+    return classificationConfigDAO;
+  }
+
+  @Override
+  public EntityToEntityMappingManager getEntityToEntityMappingDAO() {
+    return entityToEntityMappingDAO;
+  }
+
+  @Override
+  public GroupedAnomalyResultsManager getGroupedAnomalyResultsDAO() {
+    return groupedAnomalyResultsDAO;
+  }
+
+  @Override
+  public OnboardDatasetMetricManager getOnboardDatasetMetricDAO() {
+    return onboardDatasetMetricDAO;
+  }
+
+  @Override
+  public ConfigManager getConfigDAO() {
+    return configDAO;
+  }
+
+  @Override
+  public ApplicationManager getApplicationDAO() {
+    return applicationDAO;
+  }
+
+  @Override
+  public void restart() {
+    cleanup();
+    init();
```

Comment:
We need to fix the problem that init() is invoked multiple times in next PR. The init() initializes database and then DAOs. However, the DAOs will be re-used even they are reset. Therefore, those DAOs will still using the very first DB instead of new DBs.

Reply:
Yes. This operation is expensive. The origin design follows the same process. The current pr doesn't lose the performance. But we can think about how to improve the overall performance. It can be something good to have.